Python/import.c

```diff
@@ -1117,75 +1113,125 @@ list_frozen_module_names()
     return names;
 }
 
-static const struct _frozen *
-find_frozen(PyObject *modname)
+typedef enum {
+    FROZEN_OKAY,
+    FROZEN_BAD_NAME,    // The given module name wasn't valid.
+    FROZEN_NOT_FOUND,   // It wasn't in PyImport_FrozenModules.
+    FROZEN_DISABLED,    // -X frozen_modules=off (and not essential)
+    FROZEN_EXCLUDED,    // The PyImport_FrozenModules entry has NULL "code".
+    FROZEN_INVALID,     // The PyImport_FrozenModules entry is bogus.
+} frozen_status;
+
+static inline void
+set_frozen_error(frozen_status status, PyObject *modname)
+{
+    const char *err = NULL;
+    switch (status) {
```

Comment:
I wonder if these exception strings can help a user fix their problem (if they even *can* fix their problem).  Given that three enums produce the same error string, and I'm not sure whether a user will understand what a frozen module being "excluded" or "invalid" I don't know that a user will understand what's going on, let alone whether it can be fixed and how.

Reply:
Agreed.  My intent here was to preserve the current behavior.  This could definitely be approved.  I'll create a separate issue for that.

Reply:
Thanks.  Please nosy me on that separate issue.

Reply:
https://bugs.python.org/issue45379